### PR TITLE
Add Aos components

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: prod-aos-rcar.yaml [hunk 1]
```diff
@@ -166,6 +166,7 @@ components:
         - "../meta-xt-common/meta-xt-driver-domain"
         - "../meta-xt-rcar/meta-xt-rcar-fixups"
         - "../meta-xt-rcar/meta-xt-rcar-driver-domain"
+        - "../meta-xt-rcar/meta-xt-rcar-domx"
```
Comment (rshym) [inline prod-aos-rcar.yaml:216]: This change looks like part of something else. At least it's not clear how this layer relates to the commit "Update optee-os to version 3.18".

File: meta-xt-prod-aos-rcar-domf/recipes-security/optee/optee-client_git.bbappend [hunk 2]
```diff
@@ -0,0 +1,15 @@
+PV = "3.18.0+git${SRCPV}"
```
Comment (rshym) [inline meta-xt-prod-aos-rcar-domf/recipes-security/optee/optee-client_git.bbappend:1]: Misprint in commit message 'vertsion'.

File: meta-xt-prod-aos-rcar-domd/recipes-core/images/core-image-minimal.bbappend [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Enable RO rootfs
 IMAGE_FEATURES_append = " read-only-rootfs"
 
+# Xen related packages
```
Comment (rshym) [inline meta-xt-prod-aos-rcar-domd/recipes-core/images/core-image-minimal.bbappend]: This comment doesn't look relevant to the commit message.
Also, the following append contains not only "xen related packages".

File: meta-xt-prod-aos-rcar-domf/recipes-core/images/core-image-minimal.bbappend [hunk 4]
```diff
@@ -1,7 +1,13 @@
 # Enable RO rootfs
 IMAGE_FEATURES_append = " read-only-rootfs"
 
+# Xen related packages
```
Comment (rshym) [inline meta-xt-prod-aos-rcar-domf/recipes-core/images/core-image-minimal.bbappend]: Not relevant comment.

File: meta-xt-prod-aos-rcar-dom0/recipes-extended/xen/xen-tools_git.bbappend [hunk 5]
```diff
@@ -0,0 +1,5 @@
+do_install_append() {
+    # WA reduce stop timeout
+    echo "[Service]" >> ${D}${systemd_system_unitdir}/xendomains.service
+    echo "TimeoutStopSec=10" >> ${D}${systemd_system_unitdir}/xendomains.service
```
Comment (rshym) [inline meta-xt-prod-aos-rcar-dom0/recipes-extended/xen/xen-tools_git.bbappend]: I suspect that this change can be implemented as "systemd drop-in file".

File: meta-xt-prod-aos-rcar-domd/recipes-aos/aos-updatemanager/aos-updatemanager_git.bbappend [hunk 6]
```diff
@@ -0,0 +1,78 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (rshym) [inline meta-xt-prod-aos-rcar-domd/recipes-aos/aos-updatemanager/aos-updatemanager_git.bbappend:1]: Maybe "[domd] Add UM" can be squashed with the previous commit "[domf] Add UM".

File: meta-xt-prod-aos-rcar-domd/recipes-aos/aos-provfinish/aos-provfinish.bbappend [hunk 7]
```diff
@@ -0,0 +1,4 @@
+do_install_append() {
+    sed -i "s:/var/aos:/var:g" ${D}${aos_opt_dir}/provfinish.sh
```
Comment (rshym) [inline meta-xt-prod-aos-rcar-domd/recipes-aos/aos-provfinish/aos-provfinish.bbappend:2]: Here I'm not sure, but I suspect that this `provfinish.sh` was added by some 'not yet merged' commit previously. So maybe it's better to make this replacement in the original commit?
Maybe my assumption is wrong, but it seems that we are 'seding' our own file.

File: prod-aos-rcar.yaml [hunk 8]
```diff
@@ -380,20 +380,6 @@ parameters:
           AOS_BOARD_MODEL: "h3ulcb-4x2g-kf"
           AOS_BOARD_VERSION: "1.0"
 
-        components:
```
Comment (rshym) [inline prod-aos-rcar.yaml:269]: Are you sure that we do not need this layer for KF? It contains drivers, board-specific changes, and modifications to IPLs (like bl2). I'm not sure we will run on KF without this layer.
Can you confirm that this change was tested on KF?

Comment (al1img) [reply prod-aos-rcar.yaml:269]: Tested but I don't update bl2

File: prod-aos-rcar.yaml (hunk 1 above)
Comment (al1img) [reply prod-aos-rcar.yaml:216]: This meta layer was missed during initial transfer from meta-xt-prod-devel-rcar. Add by additional commit.

File: meta-xt-prod-aos-rcar-domf/recipes-security/optee/optee-client_git.bbappend (hunk 2 above)
Comment (al1img) [reply meta-xt-prod-aos-rcar-domf/recipes-security/optee/optee-client_git.bbappend:1]: fixed

File: meta-xt-prod-aos-rcar-domd/recipes-core/images/core-image-minimal.bbappend (hunk 3 above)
Comment (al1img) [reply meta-xt-prod-aos-rcar-domd/recipes-core/images/core-image-minimal.bbappend]: removed

File: meta-xt-prod-aos-rcar-domf/recipes-core/images/core-image-minimal.bbappend (hunk 4 above)
Comment (al1img) [reply meta-xt-prod-aos-rcar-domf/recipes-core/images/core-image-minimal.bbappend]: removed

File: meta-xt-prod-aos-rcar-dom0/recipes-extended/xen/xen-tools_git.bbappend (hunk 5 above)
Comment (al1img) [reply meta-xt-prod-aos-rcar-dom0/recipes-extended/xen/xen-tools_git.bbappend]: It was WA. But was fixed by adding proper dependencies to domd.service. I will drop this commit.

File: meta-xt-prod-aos-rcar-domd/recipes-aos/aos-updatemanager/aos-updatemanager_git.bbappend (hunk 6 above)
Comment (al1img) [reply meta-xt-prod-aos-rcar-domd/recipes-aos/aos-updatemanager/aos-updatemanager_git.bbappend:1]: I'll keep as separate. To copy have possibility separately cherry-pick commits to another products for example.

File: meta-xt-prod-aos-rcar-domd/recipes-aos/aos-provfinish/aos-provfinish.bbappend (hunk 7 above)
Comment (al1img) [reply meta-xt-prod-aos-rcar-domd/recipes-aos/aos-provfinish/aos-provfinish.bbappend:2]: It is part of meta-aos, an external meta-layer.

**Comment by rshym** [comment]
Acked-by: Ruslan Shymkevych <<ruslan_shymkevych@epam.com>>
